src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int count;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail != null) {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        if (head == null) {
+            head = newNode;
+        }
+        count++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size()) {
+            add(value);
+        } else if (index == 0) {
+            addToHead(value);
+        } else {
+            Node<T> source = getNodeByIndex(index);
+            Node<T> toAdd = new Node<>(source.previous, value, source);
+            source.previous.next = toAdd;
+            source.previous = toAdd;
+            count++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        list.forEach(this::add);
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexToGet(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexToGet(index);
+        Node<T> toSet = getNodeByIndex(index);
+        T oldValue = toSet.value;
+        toSet.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexToGet(index);
+        Node<T> toRemove = getNodeByIndex(index);
+        unlink(toRemove);
+        return toRemove.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        return unlink(getNodeByValue(object));
```

Comment:
`unlink` method shouldn't return any `boolean` value - it either unlinks a node or not

instead I would get rid of `getNodeByValue` method, since it is used only here and declare a method like `boolean containsNode(object)`

this ways you would be able to check if such node exists and then do unlink

benefits - code becomes more clear and you don't need to deal with null values

Reply:
I need such method getNodeByValue. Cos in public boolean remove(T object) I have to find Node (param is just Value)